ext/date/tests/005.phpt

```diff
@@ -6,30 +6,42 @@ date_default_timezone_set('UTC');
 
 $t = mktime(0,0,0, 6, 27, 2006);
 
-var_dump(idate(1,1));
-var_dump(idate(""));
-var_dump(idate(0));
+try {
+    var_dump(idate(1,1));
+} catch (\ValueError $e) {
```

Comment:
I am not keen on this change, as this is a BC break. Previously wrong values just returned `false`, which is something that `strtotime` and the older date functions have done for decades. Upgrading them to throw an exception is not a great thing to do here. It's going to cause BC breaks, even if people already carefully checked the return values. As most input to date functions will come from either user input or a database, I don't believe this is a worthy place to introduce breaks.

Instead, we should point people to the DateTime classes.